### PR TITLE
update documentation and fix errors

### PR DESCRIPTION
**Pull request by JeanCarlosChavarriaHughes**
This Pull Request is for:

* Update Documentation by adding steps to validate firmarXML endpoint
* remove plazoCredito not equal to zero validation
* added validation for codigoComercial greater than 5 items
* added validation for descuento greater than 5 items
* added new codigoComercialLinea option
* Fix error in TotalExonerado node.
* added default value to totalIVADevuelto

### CODE REVIEW

File: api/contrib/signXML/README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Modulo para firmar un XML
-Se enviar                               
+Se debe enviar
```
Comment (fdelapena) [inline api/contrib/signXML/README.md:2]: Quizás se puede aprovechar para formatear el resto del markdown: el fichero parece que estaba pensado para texto plano y podría formatearse con viñetas para las anidaciones que contiene.

File: api/contrib/firmarXML/README.md [hunk 2]
```diff
@@ -43,7 +43,34 @@ print_r($archivo);
...
+Para validar el funcionamiento de este endpoint, se debe tener un archivo p12 con el cual hacer las pruebas.
+Supongamos que el p12 se llama 070211023522.p12
+En este caso, el firmador firmarXml lo que va a hacer es recibir un downloadCode, el cual representa un archivo en una ruta local asignado a un usuario de la Api. Por lo cuál se requiere tener un usuario activo en la Api con un respectivo idUser.
+
+* Paso 1:
+Determinar el idUser que desea usar en la Api. `SELECT * from users;`
+Ejemplo, idUser = 5.
+
+* Paso 2:
+Verificar que la siguiente ruta existe en su servidor de la Api, sino existe la debe crear:
```
Comment (fdelapena) [inline api/contrib/firmarXML/README.md:55]: s/`sino`/`si no`/

Comment (JeanCarlosChavarriaHughes) [reply api/contrib/firmarXML/README.md:55]: done  Please review. https://github.com/CRLibre/API_Hacienda/pull/151
I accidentally merged this PR, so I reverted .

File: api/contrib/signXML/README.md (hunk 1 above)
Comment (JeanCarlosChavarriaHughes) [reply api/contrib/signXML/README.md:2]: done  Please review. https://github.com/CRLibre/API_Hacienda/pull/151
I accidentally merged this PR, so I reverted .

**Pull request merged**